mmdet/apis/ote/apis/detection/nncf_task.py

```diff
@@ -0,0 +1,236 @@
+# Copyright (C) 2021 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions
+# and limitations under the License.
+
+import io
+import os
+import tempfile
+import torch
+import json
+from collections import defaultdict
+from typing import Optional, Dict
+
+from ote_sdk.configuration import cfg_helper
+from ote_sdk.configuration.helper.utils import ids_to_strings
+from ote_sdk.entities.optimization_parameters import OptimizationParameters
+from ote_sdk.entities.task_environment import TaskEnvironment
+from ote_sdk.usecases.reporting.time_monitor_callback import TimeMonitorCallback
+from ote_sdk.usecases.tasks.interfaces.optimization_interface import IOptimizationTask, OptimizationType
+
+from sc_sdk.entities.datasets import Dataset, Subset
+from sc_sdk.entities.model import Model, ModelStatus, NullModel
+from sc_sdk.logging import logger_factory
+
+from mmcv.utils import Config
+from mmdet.apis import train_detector
+from mmdet.apis.ote.apis.detection.config_utils import patch_config
+from mmdet.apis.ote.apis.detection.config_utils import set_hyperparams
+from mmdet.apis.ote.apis.detection.config_utils import prepare_for_training
+from mmdet.apis.ote.apis.detection.configuration import OTEDetectionConfig
+from mmdet.apis.ote.apis.detection.configuration_enums import NNCFCompressionPreset
+
+from mmdet.apis.fake_input import get_fake_input
+from mmdet.apis.ote.apis.detection.base_task import OTEBaseTask
+from mmdet.apis.ote.extension.utils.hooks import OTELoggerHook
+from mmdet.datasets import build_dataset
+from mmdet.datasets import build_dataloader
+from mmdet.integration.nncf import check_nncf_is_enabled
+from mmdet.integration.nncf import is_state_nncf
+from mmdet.integration.nncf import wrap_nncf_model
+from mmdet.integration.nncf.config import compose_nncf_config
+
+
+logger = logger_factory.get_logger("NNCFDetectionTask")
+
+
+COMPRESSION_MAP = {
+    NNCFCompressionPreset.QUANTIZATION: "nncf_quantization",
+    NNCFCompressionPreset.PRUNING: "nncf_pruning",
+    NNCFCompressionPreset.QUANTIZATION_PRUNING: "nncf_pruning_quantization"
+}
+
+
+class NNCFDetectionTask(OTEBaseTask, IOptimizationTask):
```

Comment:
When it was decided that NNCF capabilities should be provided as a stand-alone Task implementation?